src/lib/badge/_badge-theme.scss

```diff
@@ -0,0 +1,165 @@
+@import '../core/theming/palette';
+@import '../core/theming/theming';
+@import '../core/typography/typography-utils';
+
+$mat-badge-font-size: 12px;
+$mat-badge-font-weight: 600;
+$mat-badge-default-size: 22px !default;
+$mat-badge-small-size: $mat-badge-default-size - 6;
+$mat-badge-large-size: $mat-badge-default-size + 6;
+
+// Mixin for building offset given different sizes
+@mixin _mat-badge-size($size) {
+  .mat-badge-content {
+    width: $size;
+    height: $size;
+    line-height: $size;
+  }
+
+  &.mat-badge-above {
+    .mat-badge-content {
+      top: -$size / 2;
+    }
+  }
+
+  &.mat-badge-below {
+    .mat-badge-content {
+      bottom: -$size / 2;
+    }
+  }
+
+  &.mat-badge-before {
+    margin-left: $size;
+
+    &[dir='rtl'] {
+      margin-left: 0;
+      margin-right: $size;
+    }
+
+    .mat-badge-content {
+      left: -$size;
+    }
+  }
+
+  &.mat-badge-after {
+    margin-right: $size;
+
+    &[dir='rtl'] {
+      margin-right: 0;
+      margin-left: $size;
+    }
+
+    .mat-badge-content {
+      right: -$size;
+    }
+  }
+
+  &.mat-badge-overlap {
+    &.mat-badge-before {
+      margin-left: $size / 2;
+
+      &[dir='rtl'] {
+        margin-left: 0;
+        margin-right: $size / 2;
+      }
+
+      .mat-badge-content {
+        left: -$size / 2;
+      }
+    }
+
+    &.mat-badge-after {
+      margin-right: $size / 2;
+      
+      &[dir='rtl'] {
+        margin-right: 0;
+        margin-left: $size;
+      }
+
+      .mat-badge-content {
+        right: -$size / 2;
+      }
+    }
+  }
+}
+
+@mixin mat-badge-theme($theme) {
+  $accent: map-get($theme, accent);
+  $warn: map-get($theme, warn);
+  $primary: map-get($theme, primary);
+  
+    .mat-badge-content {
+      color: mat-color($primary, default-contrast);
+      background: mat-color($primary);
+    }
+
+    &.mat-badge-accent {
+      .mat-badge-content {
+        background: mat-color($accent);
+        color: mat-color($accent, default-contrast);
+      }
+    }
+  
+    &.mat-badge-warn {
+      .mat-badge-content {
+        color: mat-color($warn, default-contrast);
+        background: mat-color($warn);
+      }
+    }
+  }
+}
```

Comment:
There's an extra closing `}` here (masked by the extra indentation in the mixin block)